drivers/hmis/app/graphql/mutations/join_household.rb

```diff
@@ -0,0 +1,103 @@
+
+    field :receiving_household, Types::HmisSchema::Household, null: false
+    field :donor_household, Types::HmisSchema::Household, null: true # Will be null if there are no remaining members
+    # todo @martha - possible that there will be multiple households with same ID? mutation should require project_id?
+    def resolve(receiving_household_id:, joining_enrollment_inputs:)
+      receiving_enrollments = Hmis::Hud::Enrollment.
+        where(household_id: receiving_household_id).
+        viewable_by(current_user).
+        includes(:project, :household)
+      access_denied! if receiving_enrollments.empty?
+
+      receiving_household = receiving_enrollments.first.household
+
+      receiving_project = receiving_enrollments.map(&:project).uniq.sole
+      access_denied! unless current_permission?(permission: :can_split_households, entity: receiving_project)
+
+      # The mutation input is a list, so convert it to a hashmap for ease of access
+      map_enrollment_id_to_relationship = joining_enrollment_inputs.map { |e| [e.enrollment_id, e.relationship_to_hoh] }.to_h
+
+      joining_enrollment_ids = map_enrollment_id_to_relationship.keys
+      joining_enrollments = Hmis::Hud::Enrollment.
+        viewable_by(current_user).
+        where(id: joining_enrollment_ids).
+        includes(:household)
+      access_denied! unless joining_enrollments.count == joining_enrollment_inputs.count
+
+      raise 'Cannot merge enrollments from another project' unless joining_enrollments.pluck(:project_pk).uniq.sole == receiving_project.id
+
+      donor_household = joining_enrollments.map(&:household).uniq.sole
+      remaining_enrollments = Hmis::Hud::Enrollment.
+        viewable_by(current_user). # Restrict to this data source
+        where(household_id: donor_household.household_id).
+        where.not(id: joining_enrollment_ids)
+      remaining_hoh = remaining_enrollments.any? { |enrollment| enrollment.relationship_to_hoh == 1 }
+      raise 'This operation would leave behind a household with no HoH, which is not allowed' unless remaining_enrollments.empty? || remaining_hoh
+
+      # If the receiving household is assigned to a unit, re-assign the joining enrollments to the same unit.
+      # This works for the present-tense, but could be improved to better accommodate past data correction.
+      units = receiving_enrollments.map { |enrollment| enrollment.active_unit_occupancy&.unit }.compact.uniq
+
+      receiving_before_state = Hmis::Hud::Enrollment.snapshot_enrollments(receiving_enrollments)
+      donor_before_state = Hmis::Hud::Enrollment.snapshot_enrollments([*joining_enrollments, *remaining_enrollments])
+
+      Hmis::Hud::Enrollment.transaction do
+        joining_enrollments.each_with_index do |enrollment, index|
+          enrollment.update!(
+            household_id: receiving_household_id,
+            relationship_to_hoh: map_enrollment_id_to_relationship[enrollment.id.to_s],
+          )
```

Comment:
Is it necessary to do two saves per enrollment? (`update!` here and `save!` below). I know `assign_unit` does some validations on household membership, but I _think_ it should work even if the household_id change hasn't been persisted yet.